test/addons/esm/test-esm.mjs

```diff
@@ -0,0 +1,50 @@
+// eslint-disable-next-line node-core/require-common-first
+import { buildType } from '../../common/index.mjs';
+import assert from 'node:assert';
+import { createRequire } from 'node:module';
+
+const require = createRequire(import.meta.url);
+
+export async function run() {
+  // binding.node
+  {
+    const bindingPath = require.resolve(`./build/${buildType}/binding.node`);
```

Comment:
Can you add a test for `main` field look up, and the `"node-addons"` exports condition with the `.node` suffix as well?